File: array/207.cpp
#include <bits/stdc++.h>

using namespace std;

using ll = long long int;

long long maximumHappinessSum(vector<int>& happiness, int k) {
    ll answer = 0;
    int decrement = 0;
    sort(happiness.begin(), happiness.end(), greater<int>());

    for (int i = 0; i < k; ++i) {
        if (i != 0) {
            answer += max(happiness[i] - decrement, 0);
        } else {
            answer += happiness[i];
        }
        decrement += 1;
    }

    return answer;
}

int main() {
    vector<int> v = {1, 2, 3};
    int k = 2;

    cout << maximumHappinessSum(v, k);

    return 0;
}

File: array/207.py
from typing import List


def maximumHappinessSum(happiness: List[int], k: int) -> int:
    happiness.sort(reverse=True)

    ans, index, decrement = 0, 0, 0

    while k > 0:
        if happiness[index] > 0:
            ans += max(happiness[index] - decrement, 0)
            decrement += 1
            index += 1
            k -= 1

    return ans


v = [2, 3, 4, 5]
k = 1
print(maximumHappinessSum(v, k))
